File: misc/aooswd/ex/memento/memento1.java
interface GeneralInterface
{
}

interface SpecificInterface
{
	public void SetState (State S);
	public State GetState ();
}


public class Memento implements GeneralInterface, SpecificInterface
{
	private State St;
	public Memento ();
	public void SetState (State S);
	public State GetState ();
}

File: misc/aooswd/ex/memento/memento1.cpp
// C++  - use friend class to implement wide interface

class Originator
{
public:
	Memento* CreateMemento ();
	void SetMemento (const Memento*);
private:
	State* _state;
};

class Memento
{
public:
	virtual ~Memento ();
private:
	friend class Originator ();
	Memento ();
	void SetState (State*);
	State* GetState ();
private:
	State* _state;
};
